lib/streamlit/connections/file_connection.py

```diff
@@ -0,0 +1,149 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from contextlib import contextmanager
+from datetime import timedelta
+from io import TextIOWrapper
+from pathlib import Path
+from typing import TYPE_CHECKING, Iterator, Optional, Union
+
+import pandas as pd
+
+from streamlit.connections import BaseConnection
+from streamlit.runtime.caching import cache_data
+
+if TYPE_CHECKING:
+    from fsspec import AbstractFileSystem, filesystem
+    from fsspec.spec import AbstractBufferedFile
+
+
+class File(BaseConnection["AbstractFileSystem"]):
```

Comment:
Hm, I think the name `Filesystem` or `FileSystem` would be better here -- `File` seems to me like it'd be a connection to a single file.

Reply:
Agreed -- I would prefer FileSystem, or FileConnection perhaps as I had titled it in my repo. The very fact that I can do:

```python
conn.read_csv('one_path')
conn.read_text('another_path')
``` 
seems to make the File naming a bit misleading, as if it's a connection to a particular file, rather than a filesystem